client/src/main/java/io/lionweb/client/inmemory/RepositoryData.java

```diff
@@ -139,8 +172,115 @@ void retrieve(String nodeId, int limit, List<SerializedClassifierInstance> retri
     }
     retrieved.add(node);
     if (limit > 0) {
-      node.getChildren().forEach(c -> retrieve(c, limit - 1, retrieved));
-      node.getAnnotations().forEach(a -> retrieve(a, limit - 1, retrieved));
+      node.getChildren()
+          .forEach(
+              c -> {
+                try {
+                  retrieve(c, limit - 1, retrieved);
+                } catch (Exception e) {
+                  throw new RuntimeException("Unable to retrieve child of " + node, e);
+                }
+              });
+      node.getAnnotations()
+          .forEach(
+              a -> {
```

Comment:
(naming) `annotation[Id]` instead of `a`?

Reply:
Done -> `annotationId`